app/code/core/Mage/GoogleAnalytics/Block/Ga.php

```diff
@@ -398,14 +397,16 @@ protected function _getOrdersTrackingCodeAnalytics4()
                     'item_id' => $_product->getSku(),
                     'item_name' => $_product->getName(),
                     'price' => number_format($_product->getFinalPrice(), 2),
+
+                    'quantity' => intval($productInCart->getQty()),
                 ];
                 if ($_product->getAttributeText('manufacturer')) {
                     $_item['item_brand'] = $_product->getAttributeText('manufacturer');
                 }
-                if ($_product->getCategoryIds()) {
-                    $_lastCat = end($_product->getCategoryIds());
-                    $_cat = Mage::getModel('catalog/category')->load($_lastCat);
-                    $_item['item_category'] = $_cat->getName();
+
+                $item_category = Mage::helper('googleanalytics')->getLastCategoryName($_product);
```

Comment:
openmage doesn't use the "item_category" notation, but the "itemCategory" one, which I also don't love that much but... it is what it is...

could you use my code from https://github.com/OpenMage/magento-lts/pull/3323 or it doesn't work for you?

Reply:
the problem is variable name or wrap in helper function getLastCategoryName()?
your code works but is for a well organized code

Reply:
variable names

Reply:
fixed